### DISCUSSION TITLE
How to change response model for "422 Unprocessable Entity" in OpenAPI docs

### DISCUSSION THREAD

**Discussion by kokumura** [fastapi/fastapi#8134] [Questions] [answered]
**Description**
I am trying to return my own response model for `422 Unprocessable Entity` error.

I added a custom exception handler to my app, and add my response model type to `responses` parameter of `router.post()`.

```python
def MyValidationError(BaseModel):
  message: str

@app.exception_handler(RequestValidationError)
def handle_validation_error(request, exc) -> JSONResponse:
    return JSONResponse(
        status_code=HTTP_422_UNPROCESSABLE_ENTITY,
        content=MyValidationError('invalid parameter').dict(),
    )

@router.post("/user", responses={422: {"model": MyValidationError}})
def foo(user_id: str, user_name: str):
  ...
```

`/user` works as I expected.

But `/openapi.json` still refers the default HTTPValidationError model.

```
{
  ...
  "paths": {
    "/user": {
      "post": {
        "responses": {
          "422": {
            "description": "Validation Error",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/HTTPValidationError"
                }
```

I found the `HTTPValidationError` schema sesms to be hardcoded in [fastapi/openapi/utils.py](https://github.com/tiangolo/fastapi/blob/7244e4b61290742993e55caaf6f334634e7a9503/fastapi/openapi/utils.py#L214).

I expected that response models in openapi.json should reflect either custom exception handler for `RequestValidationError` or `responses` parameter for routes.

Is this a bug?
If not, is there any way to set correct response models for status `422` in documents?

**Comment by tiangolo** [comment, marked as answer]
Yes, as of now they are still hardcoded. I haven't defined a way to add other models that could be returned by exception handlers.

For now, the most straightforward way would be to override the OpenAPI: https://fastapi.tiangolo.com/tutorial/extending-openapi/

**Comment by kokumura** [comment]
OK. I got it.
I'll try to modify the OpenAPI schema.
Thanks!

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue.

**Comment by rozeappletree** [comment]
> Yes, as of now they are still hardcoded. I haven't defined a way to add other models that could be returned by exception handlers.
> 
> For now, the most straightforward way would be to override the OpenAPI: https://fastapi.tiangolo.com/tutorial/extending-openapi/

link not working!

**Comment by phy25** [comment]
@rakesh4real https://fastapi.tiangolo.com/advanced/extending-openapi/

**Comment by 243f6a8885a308d313198a2e037** [comment]
This appears to have been fixed in #437.
